### ISSUE TITLE
SC2039 wrongly trigged on $'..' C-like strings

### ISSUE THREAD

**Issue by leagris** [koalaman/shellcheck#1678]
#### For bugs
- Rule Id: SC2039
- My shellcheck version: online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env sh
printf 
```

#### Here's what shellcheck currently says:
```
Line 2	SC2039: In POSIX sh,  $'..'
```

#### Here's what shellcheck currently says:
```
Line 2:
printf $'%s\n' 'Hello world!'
       ^-- SC2039: In POSIX sh, $'..' is undefined.
```
#### Here's what I wanted or expected to see:

shellcheck should not complain on C-like strings `$'..'` because it is an accepted POSIX standard, and is supported by `ash`, `dash`, `ksh93`, `zsh` and probably a whole lot if not all modern POSIX-compliant shells.

#### Other legitimate use:

```sh
#!/usr/bin/env sh

IFS='$\n'
```

See: [SC2141](https://github.com/koalaman/shellcheck/wiki/SC2141)

**Comment by koalaman** [comment]
I checked POSIX.1-2017 aka [Open Group Base Specifications Issue 7](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html) and couldn't find a mention of it. Are you saying that it has been accepted for an upcoming version of the POSIX standard?

**Comment by leagris** [comment]
See:
- http://austingroupbugs.net/view.php?id=249
- http://mywiki.wooledge.org/Bashism
> Historically $' ' was not defined by POSIX through 2008, but has been accepted for the next version. http://austingroupbugs.net/view.php?id=249. Originated in ksh93, also supported by zsh, mksh, FreeBSD and Busybox sh though with variations

**Comment by tuffnatty** [comment]
The change has been applied to the working version of the standard: https://www.mail-archive.com/austin-group-l@opengroup.org/msg10365.html .
The warning that shellcheck produces has changed to SC3003 since the bug creation.

**Comment by GrantGryczan** [comment]
Because of this bug, I just add this to every `.shellcheckrc`:

```
# Fix koalaman/shellcheck#1678.
disable=SC3003
```
